src/Waives.Pipelines/ObservableProcessExtension.cs

```diff
@@ -32,14 +36,14 @@ internal static class ProcessExtension
             return results.Select(r => r.Document);
         }
 
-        private static async Task<DocumentError> OnProcessingErrorAsync<TDocument>(ProcessingError<TDocument> error)
+        private static async Task<DocumentError> OnProcessingErrorAsync<TDocument>(ProcessingError<TDocument> error, CancellationToken cancellationToken = default)
         {
             // Try coercing the document in error to a Waives Document. If it is, it
             // has been created in the platform and must be deleted before proceeding.
             var waivesDocument = error.Document as WaivesDocument;
             if (waivesDocument != null)
             {
-                await waivesDocument.DeleteAsync().ConfigureAwait(false);
+                await waivesDocument.DeleteAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
We probably don't want these requests to be cancelled - we'd still want the documents to be cleared up after an error.